backend_py/primary/primary/services/sumo_access/_arrow_table_loader.py

```diff
@@ -114,23 +120,54 @@ async def get_aggregated_multiple_columns_async(
         self,
         column_names: list[str],
     ) -> pa.Table:
-        """Fetches multiple columns async and aggregates them into a single Arrow table"""
+        """
+        Fetches aggregated table for multiple columns async and assembles them into a single Arrow table
+        """
+        if not column_names:
+            raise InvalidParameterError(
+                f"Cannot fetch aggregated tables for empty column list: {self._make_req_info_str()}", Service.SUMO
+            )
 
+        # Fetch the aggregated table for each column
         async with asyncio.TaskGroup() as tg:
-            tasks = [
-                tg.create_task(self.get_aggregated_single_column_async(column_name)) for column_name in column_names
-            ]
-
-        table_arr: list[pa.Table] = [task.result() for task in tasks]
+            column_name_task_dict = {
```

Comment:
Usage of a dict here (instead of just a list) is a bit confusing. 
Is there a specific reason?